client/blocks/comment-detail/docs/comment-faker.jsx

```diff
@@ -0,0 +1,98 @@
+/**
+ * External dependencies
+ */
+import React, { Component } from 'react';
+import { keyBy, omit } from 'lodash';
+
+/**
+ * `CommentFaker` is a HOC to easily test the Comments Management without the necessity of real data or actions.
+ *
+ * It takes the comments passed via props (with mock data, Redux-connected, etc.) and puts them in its internal state.
+ * Then passes them and various actions (status and like setters) to the actual comment list component.
+ *
+ * Once data and actions will be completely Reduxified, it will be enough to remove the `CommentFaker` call
+ * and use the `mapDispatchToProps` actions instead, leaving this component still useful for testing purposes.
+ */
+export const CommentFaker = WrappedCommentList => class extends Component {
+	state = {
+		comments: {},
+	};
+
+	componentWillMount() {
+		if ( this.props.comments.length ) {
+			this.setState( { comments: keyBy( this.props.comments, 'ID' ) } );
+		}
+	}
+
+	componentWillReceiveProps( nextProps ) {
+		if ( ! this.props.comments.length ) {
+			this.setState( { comments: keyBy( nextProps.comments, 'ID' ) } );
+		}
+	}
+
+	deleteCommentPermanently = commentId => this.setState( { comments: omit( this.state.comments, commentId ) } );
+
+	setCommentLike = ( commentId, likeValue ) => {
+		const comment = this.state.comments[ commentId ];
+
+		// If like changes to true, also approve the comment
+		this.setState( {
+			comments: {
+				...this.state.comments,
+				[ commentId ]: {
+					...comment,
+					i_like: likeValue,
+					status: likeValue ? 'approved' : comment.status,
+				}
+			},
+		} );
+	}
+
+	setCommentStatus = ( commentId, status ) => {
+		const comment = this.state.comments[ commentId ];
+
+		// If the comment is not approved anymore, also remove the like, otherwise keep its previous value
+		this.setState( {
+			comments: {
+				...this.state.comments,
+				[ commentId ]: {
+					...comment,
+					i_like: 'approved' === status ? comment.i_like : false,
+					status,
+				}
+			},
+		} );
+	}
+
+	toggleCommentLike = commentId => {
+		const comment = this.state.comments[ commentId ];
+		const likeValue = ! comment.i_like;
+
+		// If like changes to true, also approve the comment
+		this.setState( {
+			comments: {
+				...this.state.comments,
+				[ commentId ]: {
+					...comment,
+					i_like: likeValue,
+					status: likeValue ? 'approved' : comment.status,
+				}
+			},
+		} );
+	}
+
+	render() {
+		return (
+			<WrappedCommentList
+				{ ...this.props }
+				comments={ this.state.comments }
```

Comment:
> don't really care if it's an array or an object

the big difference here is going to be whether or not the component expects the comments to come in with a specific sort order. I doubt it since the component will likely perform its own sort based on which buttons are clicked. in the case of an object we don't know what `values()` or `map()` will be - it should end up in the order with which the keys were added to the object. with a list the sort order is explicit unless we manually sort it - position in the list.